src/attached_probe.h

```diff
@@ -81,7 +82,7 @@ class AttachedProbe
   void cache_progfd(void);
 
   Probe &probe_;
-  std::tuple<uint8_t *, uintptr_t> func_;
+  BpfProgram &prog_;
```

Comment:
Please correct me if I'm wrong, but this looks to me like a dangling reference.

BpfProgram is created on the stack in `BPFtrace::attach_probe` from the return value of `BpfProgram::CreateFromBytecode`. Then a reference to this stack value is stored here in AttachedProbe.

Reply:
You're right, good catch. Let's just store `BpfProgram` directly inside `AttachedProbe` as it's where it belongs.